File: MAP(Advanced Programming Methods)/lab3/src/Model/Expression/VariableExpression.java
package Model.Expression;
import Exceptions.*;
import Model.ADT.*;
public class VariableExpression implements Expression {

    private String key;


    public VariableExpression(String key)
    {
        this.key=key;

    }

    public int eval(IDictionary<String,Integer> dict)
    {
        if(dict.contains(key))
            return dict.get(this.key);
        throw new VariableExpressionException("Key not in SymbolTable");
    }
    @Override
    public String toString()
    {
        return ""+key;
    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Expressions/VariableExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using LabAssig7.Model.ADT;
namespace LabAssig7.Model.Expressions
{
    public class VariableExpression:Expression
    {
        private string varName;

        public VariableExpression(string varName)
        {
            this.varName = varName;
        }

        public int Eval(ISymbolTable<string,int> dict)
        {
            return dict.GetValue(this.varName);
        }

        public override string ToString()
        {
            return this.varName;
        }
    }
}
